# topic names
## subscribing topics
odom_topic: /groundtruth_odom
ref_path_topic: /move_base/NavfnROS/plan
collision_costmap_topic: /move_base/local_costmap/costmap
distance_error_map_topic: /distance_error_map
ref_yaw_map_topic: /ref_yaw_map
## publishing topics
control_cmd_vel_topic: /cmd_vel
mppi_absvel_topic: /mppi/cmd/absvel
mppi_vx_topic: /mppi/cmd/vx
mppi_vy_topic: /mppi/cmd/vy
mppi_omega_topic: /mppi/cmd/omega
calc_time_topic: /mppi/calc_time
mppi_overlay_text_topic: /mppi/overlay_text
mppi_optimal_traj_topic: /mppi/optimal_traj
mppi_sampled_traj_topic: /mppi/sampled_traj
mppi_eval_msg_topic: /mppi/eval_info

# navigation params
navigation:
  xy_goal_tolerance: 0.5 # [m]
  yaw_goal_tolerance: 3.14 # [rad]

# target_system params
target_system:
  l_f: 0.5 # [m]
  l_r: 0.5 # [m]
  d_l: 0.5 # [m]
  d_r: 0.5 # [m]
  tire_radius: 0.2 # [m]

# mode selector params
mode_selector:
  yaw_error_threshold: 0.5 # [rad]
  dist_error_threshold: 0.5 # [m]

# controller params
controller:
  common: # common parameters
    control_interval: 0.05 # [s]
    prediction_horizon: 30 # [steps]
    step_len_sec: 0.033 # [s]

  mode1: # mode one: mppi_3d
    name: "[mode 1] mppi_3d_b"
    num_samples: 3000 # number of samples
    param_exploration: 0.1
    param_lambda: 200.0
    param_alpha: 0.975
    sigma: [0.55, 0.55, 0.96] # noise for [vx, vy, omega]
    reduce_computation: false # if true, noise sampling is done only once and the same noise is used for all processes.
    weight_cmd_change: [0.0, 0.0, 0.0] # penalty weight for variation of [vx, vy, omega]
    weight_vehicle_cmd_change: [1.4, 1.4, 1.4, 1.4, 0.1, 0.1, 0.1, 0.1] # penalty weight for variation of [fl_steer, fr_steer, rl_steer, rr_steer, fl_vel, fr_vel, rl_vel, rr_vel]
    reference_velocity: 2.0 # [m/s]
    weight_velocity_error: 10.0
    weight_angular_error: 30.0
    weight_collision_penalty: 50.0
    weight_distance_error_penalty: 40.0
    weight_terminal_state_penalty: 10.0
    use_sg_filter: true # set true to use Savitzky-Golay filter for smoothing the control input
    sg_filter_half_window_size: 4 # value in the range of 1 ~ (prediction_horizon - 1) is allowed.
    sg_filter_poly_order: 6

  mode2: # mode two: mppi_4d
    name: "[mode 2] mppi_4d"
    num_samples: 3000 # number of samples
    param_exploration: 0.1
    param_lambda: 100.0
    param_alpha: 0.975
    sigma: [0.78, 0.78, 1.00, 1.00] # noise for [fl_steer, rr_steer, fl_vel, rr_vel]
    reduce_computation: false # if true, noise sampling is done only once and the same noise is used for all processes.
    weight_cmd_change: [0.0, 0.0, 0.0] # penalty weight for variation of [vx, vy, omega]
    weight_vehicle_cmd_change: [1.4, 1.4, 1.4, 1.4, 0.1, 0.1, 0.1, 0.1] # penalty weight for variation of [fl_steer, fr_steer, rl_steer, rr_steer, fl_vel, fr_vel, rl_vel, rr_vel]
    reference_velocity: 2.0 # [m/s]
    weight_velocity_error: 10.0
    weight_angular_error: 30.0
    weight_collision_penalty: 50.0
    weight_distance_error_penalty: 40.0
    weight_terminal_state_penalty: 10.0
    use_sg_filter: true # set true to use Savitzky-Golay filter for smoothing the control input
    sg_filter_half_window_size: 4 # value in the range of 1 ~ (prediction_horizon - 1) is allowed.
    sg_filter_poly_order: 6
